File: Greedy/p4.py
# Solution 1
n, m = map(int, input().split())
cnt = 0
while n != 1:
    if n % m != 0:
        cnt += 1
    else:
        n //= m
        cnt += 1
print(cnt)

#Solution 2
n, m = map(int, input().split())
cnt = 0
tmp = 0
while(True):
    tmp = (n // m) * m
    cnt += n - tmp
    n = tmp
    if(n < m):
        break
    n = n // m
    cnt += 1
cnt += n - 1
print(cnt)

File: Greedy/p4.cpp
// Solution 1
#include <bits/stdc++.h>
using namespace std;
int main(){
    int n, m;
    int cnt = 0;
    cin >> n >> m;
    while(n!=1){
        if(n % m != 0){
            n -= 1;
            cnt ++;
        }
        else{
            n /= m;
            cnt++;
        }
    }
    cout << cnt;
}
// Solution 2
#include <bits/stdc++.h>
using namespace std;
int n, m, tmp, cnt;
int main(){
    cin >> n >> m;
    while(true){
        tmp = (n / m) * m;
        cnt += n - tmp;
        n = tmp;
        if(n < m){
            break;
        }
        cnt ++;
        n = n / m;
    }
    cnt += n - 1;
    cout << cnt;

}